lib/spandex_datadog/api_server.ex

```diff
@@ -144,47 +147,51 @@ defmodule SpandexDatadog.ApiServer do
           agent_pid: agent_pid
         } = state
       ) do
-    all_traces = [trace | waiting_traces]
+    :telemetry.span([:spandex_datadog, :handle_call, :send_trace], %{trace: trace, state: state}, fn ->
```

Comment:
I need to give this some more thought, but one initial thing here is that I don't love the name of this event because I feel like the `:handle_call` is an internal detail that should be replaced with a more-permanent name, but I'm not sure what to call it at the moment. `:send_trace` above makes sense because that's the public API that's being called to "send" the trace from your app code into Spandex, but I feel like this needs some "official" name like `:forward_trace` for actually sending the trace from Spandex's buffer to Datadog. 🤔

Reply:
I'm also struggling with the thought that one of these events is fired from the caller's process and the other fires in the GenServer process. That's probably not bad but I'm trying to think through the ramifications.

Reply:
If it helps this PR get merged I can remove the telemetry spans from inside the GenServer. I only *really* need to capture the client telemetry.

Reply:
I started working on a branch to re-arrange things a little so that it would make sense to have telemetry for the client side like you mentioned, but also telemetry from inside the GenServer for enqueuing a trace and for flushing a batch of traces. We should be able to get it all sorted out today/tomorrow I think.

Reply:
Honestly, if it turns out that the client side of things is becoming a bottleneck, I have a few improvements for this flow I've already been working on. I want to prove that this is an issue before I submit them as a PR though.

Reply:
Yeah, I think we should go ahead and ship some minimal telemetry on the client side then, and we can separately clean up/optimize/add more telemetry afterword.

The only think holding up the basic telemetry around the client call is some docs about it, which I can help with. Primarily, I want to make sure we document the fact that this `:telemetry.span` would fire after it's too late to update the current trace, because the trace has already been sent to the APIServer, so you won't be able to include this time as a span in the overall trace unless you `continue_trace` and send a separate trace with only this span in it. If you did _that_ though, you'd have to be careful to not set `batch_size` to `1` or else it will infinitely send new traces of it sending traces.

Reply:
Ok @keathley, I removed the telemetry from inside the GenServer and added some docs / tests.
I believe this still does what you intended, but please let me know if I missed something!